txpool/txpool.go

```diff
@@ -837,22 +838,41 @@ func (p *TxPool) addTx(origin txOrigin, tx *types.Transaction) error {
 		return ErrTxPoolOverflow
 	}
 
+	slotsAllocated := slotsRequired(tx)
```

Comment:
I think its safe to remove check
```
if slotsRequired(tx) > slotsFree {
		return ErrTxPoolOverflow
	}
```

from the lines 845-848 and 

```
	// populate currently free slots
	slotsFree := p.gauge.freeSlots()
```
from the lines 790-791

and 

`slotsFree += slotsRequired(oldTxWithSameNonce) // add old tx slots` from the line 831

```
if slotsAllocated > slotsFreed {
		slotsIncreased = slotsAllocated - slotsFreed
		if !p.gauge.increaseWithinLimit(slotsIncreased) {
			return ErrTxPoolOverflow
		}
	}
```
check if enough i think

Otherwise, this pr looks good!

Reply:
I just removed them. Thank you.